plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -121,6 +121,22 @@ plugin = GHC.defaultPlugin { GHC.pluginRecompile = GHC.flagRecompile
              : pluginPass
              : rest
 
+{- Note [GHC.sm_pre_inline]
+We run a GHC simplifier pass before the plugin, in which we turn on `sm_pre_inline`. The
```

Comment:
I'm kind of confused here. We set `sm_inline=False`, and this new `sm_pre_inline` flag only appeared in GHC 9.2. Did they split `PreInlineUnconditional` into its own pass or something?

Reply:
I don't think `sm_inline` affects PreInlineUnconditionally, rather it only affects call site inlining. In GHC 8, PreInlineUnconditionally is only controlled by `Opt_SimplPreInlining` in DynFlags, and it is on by default.

Reply:
I guess... I still feel confused after reading this? Perhaps we need a note with a slightly broader purpose that explains all the inliner settings we're setting and why, which sadly probably requires explaining a bit more of what GHC does? it seems like you know this already, so maybe we can just write it down

Reply:
@michaelpj Please see if this is clearer.